<launch>

  <group ns="camera">
    <node pkg="libuvc_camera" type="camera_node" name="mycam">
      <!-- Parameters used to find the camera -->
      <param name="vendor" value="0x046d"/>
      <param name="product" value="0x0825"/>
      <param name="index" value="0"/>

      <!-- Image size and type -->
      <param name="width" value="1280"/>
      <param name="height" value="960"/>
      <!-- choose whichever uncompressed format the camera supports: -->
      <param name="video_mode" value="mjpeg"/> <!-- or yuyv/nv12/mjpeg -->
      <param name="frame_rate" value="30"/>

      <param name="timestamp_method" value="start"/> <!-- start of frame -->
      <param name="camera_info_url" value="file:///tmp/cam.yaml"/>

      <param name="auto_exposure" value="3"/> <!-- use aperture_priority auto exposure -->
      <param name="auto_white_balance" value="false"/>
    </node>
  </group>

<node pkg="tf2_ros" type="static_transform_publisher" name="link1_broadcaster" args="0 0 0 0 0 0 1 base camera" />

  <!-- <include file="$(find kinect2_bridge)/launch/kinect2_bridge.launch" /> -->

  <node pkg="success_ros_aruco" type="success_ros_aruco_node" name="aruco_node" output="screen">
   
    <param name="marker_dictonary" value="TAG36h11"/>
    
    <param name="marker_size" value="0.034"/>
    <param name="rate" value="30"/>

    <remap from="image" to="/camera/image_raw"/>
    <remap from="camera_info" to="/camera/camera_info"/>

    <remap from="markers" to="/markersAruco"/>
  </node>

</launch>
